payments-core/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -51,18 +60,100 @@ internal abstract class BaseAddPaymentMethodFragment(
         val viewBinding = FragmentPaymentsheetAddPaymentMethodBinding.bind(view)
         addPaymentMethodHeader = viewBinding.addPaymentMethodHeader
 
-        addPaymentMethodFragment()
+        val config = requireNotNull(
+            arguments?.getParcelable<FragmentConfig>(
+                BaseSheetActivity.EXTRA_FRAGMENT_CONFIG
+            )
+        )
+
+        val paymentMethods = config.stripeIntent.paymentMethodTypes.mapNotNull {
+            SupportedPaymentMethod.fromCode(it)
+        }.filter { it == SupportedPaymentMethod.Card }
+
+        if (paymentMethods.isEmpty()) {
+            sheetViewModel.onFatal(
+                IllegalArgumentException(
+                    "None of the requested payment methods" +
+                        " (${config.stripeIntent.paymentMethodTypes})" +
+                        " matches the supported payment types" +
+                        " (${SupportedPaymentMethod.values().toList()})"
+                )
+            )
+            return
+        }
+
+        viewBinding.googlePayDivider.setText(
+            if (paymentMethods.contains(SupportedPaymentMethod.Card) && paymentMethods.size == 1) {
+                R.string.stripe_paymentsheet_or_pay_with_card
+            } else {
+                R.string.stripe_paymentsheet_or_pay_using
+            }
+        )
+
+        if (paymentMethods.size > 1) {
+            setupRecyclerView(viewBinding, paymentMethods)
+        }
```

Comment:
Is there a problem with a recycler view of one element?  Might be nice to just keep it simple the same path regardless of the number of items in the list.